infrastructure/terraform/modules/hw-tf-module-cluster-resources/keycloak.tf

```diff
@@ -0,0 +1,180 @@
+/*
+  Keycloak is an open-source identity provider.
+  We use it for managing users, organizations and logins.
+
+  cf. https://www.keycloak.org/
+      https://www.keycloak.org/guides
+      https://www.keycloak.org/docs/latest/server_admin/index.html
+*/
+
+//
+// Variables
+//
+
+variable "keycloak_image_tag" {
+  type = string
+  description = "Which tag of our custom keycloak image should be used?"
+  default = "edge"
+}
+
+variable "keycloak_hostname" {
+  type = string
+  description = "What publicly accessible domain should kc expect to be reachable at? Also used for the ingress. Must be a domain, without path and port."
+  default = "sso.helpwave.de"
+}
+
+//
+// Locals
+//
+
+locals {
+  keycloak_database_name = "keycloak"
+
+  // TODO: use secrets when we swap out db
+  keycloak_database_user = "keycloak"
+  keycloak_database_password = "keycloak"
+}
+
+//
+// Helm Release
+//
+
+// TODO: Move away from one PostgreSQL-Server per service.
+// Database
+resource "helm_release" "keycloak_db" {
+  name  = "keycloak-postgres"
+  repository = "https://charts.bitnami.com/bitnami"
+  chart = "postgresql"
+  wait = true
+
+  namespace = "keycloak"
+  create_namespace = true
+
+  set {
+    name  = "auth.enablePostgresUser"
+    value = true
+  }
+
+  set {
+    name  = "auth.database"
+    value = local.keycloak_database_name
+  }
+
+  set {
+    name  = "auth.username"
+    value = local.keycloak_database_user
+  }
+
+  set {
+    name  = "auth.password"
+    value = local.keycloak_database_password
+  }
+
+  set {
+    name  = "auth.postgresPassword"
+    value = local.keycloak_database_password
+  }
+}
+
+// keycloak operator + CRD + Ingress
+resource "helm_release" "keycloak" {
+  name  = "keycloak"
+  chart = "../../../charts/keycloak"
+  wait = true
+
+  depends_on = [
+    helm_release.apisix,
```

Comment:
Why does Keycloak depend on APISIX?

```suggestion
```

Reply:
Not now, but later this chart will deploy a CRD for routing. But for that the CRD definition needs to exist. Hence the dependence.